tests/test-tokenizer-1.cpp

```diff
@@ -91,8 +87,8 @@ int main(int argc, char **argv) {
                 return 2;
             }
         } else {
-            if ((vocab_type(ctx) == "spm" && i <= 258) ||
-                (vocab_type(ctx) == "bpe" && (i == 0 || i >= 100000))) {
+            // TODO: needs access to token types
+            if (0 <= i && i < 259) {
```

Comment:
We can expose token types through the C-style API

e.g.:

```c++
LLAMA_API bool llama_token_is_normal(llama_token id);
...

Would that be enough?
```

Reply:
Yes, that would be fine for me.